lib/atom/decoration.js

```diff
@@ -115,6 +117,23 @@ class BareDecoration extends React.Component {
       return;
     }
 
+    // delay decoration creation when it's a gutter type;
+    // instead wait for the Gutter to be added to the editor first
+    if (this.props.type === 'gutter') {
+      if (!this.props.gutterName) {
+        throw new Error('You are trying to decorate a gutter but did not supply gutterName prop.');
+      }
+      this.props.editorHolder.map(editor => {
+        this.gutterSub = editor.observeGutters(gutter => {
+          if (gutter.name === this.props.gutterName) {
+            this.createDecoration();
+          }
+        });
+        return null;
+      });
+      return;
```

Comment:
nit: would it make sense to use `forEach` instead of `map` here to avoid having to return null?

Reply:
We'd have to implement `.forEach()` on `RefHolder` first (editorHolder isn't an Array). eslint goes purely by the method name, not the receiver type, so it doesn't know the difference. Just a RefHolder API thing that i've never gotten around to smoothing over.

Reply:
ah ok cool makes sense!